main.cpp

```diff
@@ -213,191 +334,379 @@ struct Computer
+    }
+}
+
+void Turntable::rotatePlatter()
+{
+    float speedDifference = platterRpm - static_cast< float >(speedSelection);
+    if (std::abs(speedDifference) > 0.5f)
+    {
+        platterRpm -= 0.1f*(speedDifference);
+    }
+}
+
+float grooveAmplitude()
+{
+    return static_cast<float>(rand()) / static_cast<float>(RAND_MAX); // Amplitude between 0.0f and 1.0f
+}
+
+
 struct CassetteDeck
 {
-    // cassette compartment open
     bool isCompartmentOpen = false;
-    // tape speed
     float tapeSpeedCmPerSecond = 4.76f;
-    // current at the play head
+    float reelSpeed;
+    float speedCoefficient = 1.124f;
     double playHeadCurrentMilliAmps = 0.83;
-    // control button pressed
     std::string controlButtonPressed = "Play";
-    // counter value
     int counter = 354;
+    bool isForward = true;
+    
+    int readMagneticInformation(); 
 
-    // 3 things it can do:
-    // read magnetic information 
-    double readMagneticInformation(); // returns a current value in mA
-
-    // rotate reel
-    bool rotateReel(bool isForward); // returns true if the reel rotates
+    void rotateReel(); 
 
-    // reverse direction
-    bool reverseDirection(); // returns true if the direction can be reversed
+    void reverseDirection(); 
 };
 
+int CassetteDeck::readMagneticInformation()
+{
+    return rand();
+}
+
+void CassetteDeck::rotateReel()
+{
+    
+    if(!isForward)
+    {
+        speedCoefficient = -speedCoefficient;
+    }
+    
+    reelSpeed = tapeSpeedCmPerSecond * speedCoefficient;
+}
+
+void CassetteDeck::reverseDirection()
+{
+    isForward = !isForward;
+}
+
 struct Amplifier
 {
-    // volume button value
     float volumeButtonValue = 11.1f;
-    // input signal voltage
+    float maxVolume = 100.0f;
     double inputSignalVoltageVolts = 0.0;
-    // speaker output voltage
     double speakerOutputVoltageVolts = 0.0;
-    // source selected
     std::string sourceSelected = "turntable";
-    // equalizer preset
     std::string equalizerPreset = "jazz";
 
-    // 3 things it can do:
-    // change volume
-    float changeVolume(bool up); // returns new volume
-    // output sound
-    float outputSound(float volume); // returns a signal voltage
-    // change source
-    bool changeSource(std::string newSource); // returns true if the change is successful
+    void changeVolume(bool up); 
+
+    float outputSound(); 
+
+    void changeSource(std::string newSource);
 };
 
+void Amplifier::changeVolume(bool up)
+{
+    float volumeDelta = 0.1f;
+    if (up)
+    {
+        volumeButtonValue += volumeDelta;
+        if (volumeButtonValue > maxVolume)
+        {
+            volumeButtonValue = maxVolume;
+        }
+    }
+    else
+    {
+        volumeButtonValue -= volumeDelta;
+        if (volumeButtonValue < 0)
+        {
+            volumeButtonValue = 0;
+        }
+    }
+}
+
+float Amplifier::outputSound()
+{
+    return static_cast<float>(rand()) / static_cast<float>(RAND_MAX) * volumeButtonValue / maxVolume;
```

Comment:
Casting has not been covered yet.  
Remove this cast.
Change your code so that no warnings are generated **without** using casting.
- [ ] revise